checker/check-new-request-non-path-parameter.go

```diff
@@ -28,18 +28,23 @@ func NewRequiredRequestNonPathParameterCheck(diffReport *diff.Diff, operationsSo
 				for _, paramName := range paramItems {
 					for _, param := range operationItem.Revision.Parameters {
 						if param.Value.Name == paramName {
-							if param.Value.Required {
-								source := (*operationsSources)[operationItem.Revision]
-								result = append(result, BackwardCompatibilityError{
-									Id:          "new-required-request-parameter",
-									Level:       ERR,
-									Text:        fmt.Sprintf(config.i18n("new-required-request-parameter"), ColorizedValue(paramLocation), ColorizedValue(paramName)),
-									Operation:   operation,
-									OperationId: operationItem.Revision.OperationID,
-									Path:        path,
-									Source:      source,
-								})
+							id := "new-required-request-parameter"
+							level := ERR
+							if !param.Value.Required {
+								id = "new-optional-request-parameter"
+								level = INFO
 							}
```

Comment:
cover both required (ERR level) and optional (INFO level) request non-path params